src/ngx_http_lua_socket_tcp.c

```diff
@@ -4891,6 +4967,7 @@ ngx_http_lua_socket_tcp_upstream_destroy(lua_State *L)
         return 0;
     }
 
+    luaL_unref(L, LUA_REGISTRYINDEX, u->bind_tbl_ref);
```

Comment:
I think we also need to call `luaL_unref` in other connection closing handlers like `ngx_http_lua_socket_keepalive_close_handler`. Otherwise there is a memory leak in the Lua VM.